xonsh/commands_cache.py

```diff
@@ -100,12 +101,21 @@ def all_commands(self):
         if cache_valid:
             return self._cmds_cache
         allcmds = {}
+        cnt = 0
         for path in reversed(path_immut):
             # iterate backwards so that entries at the front of PATH overwrite
             # entries at the back.
             for cmd in executables_in(path):
+                cnt += 1
```

Comment:
Rather than having a counter, you can just check the len(allcmds) after the loop.